### PR TITLE
admin have no special rights on users' entries

### PR DESCRIPTION
**Pull request by ArtificialOwl**
remove admin's right to modify users' external filesystem configuration.

### CODE REVIEW

File: apps/files_external/lib/Controller/AjaxController.php [hunk 1]
```diff
@@ -104,17 +90,13 @@
...
-
-		// Non-admins can only edit their own credentials
-		$allowedToEdit = ($this->groupManager->isAdmin($currentUser->getUID()) || $currentUser->getUID() === $uid);
-
-		if ($allowedToEdit) {
-			$this->globalAuth->saveAuth($uid, $user, $password);
-			return true;
-		} else {
+	public function saveGlobalCredentials(string $uid, string $user, string $password): bool {
+		if ($this->userSession->getUser()->getUID() !== $uid) {
```
[the remark this answers is not in the document]
Comment (ArtificialOwl) [reply apps/files_external/lib/Controller/AjaxController.php:94]: getUser() should not be null at this point

**Comment by ArtificialOwl** [comment]
failed test are not related

**Review by icewind1991 — APPROVED** [review]

**Review by AndyScherzinger — APPROVED** [review]

File: apps/files_external/tests/Controller/AjaxControllerTest.php [hunk 2]
```diff
@@ -141,18 +130,13 @@ public function testSaveGlobalCredentialsAsAdminForSelf() {
 	public function testSaveGlobalCredentialsAsNormalUserForSelf() {
 		$user = $this->createMock(IUser::class);
 		$user
-			->expects($this->exactly(2))
```
Comment (icewind1991) [inline apps/files_external/tests/Controller/AjaxControllerTest.php:144]: I'm in favor of just removing these count checks unless the number of times something is called is actually relevant to the test

Comment (ArtificialOwl) [reply apps/files_external/tests/Controller/AjaxControllerTest.php:144]: Thanks! and I thought I was the only one

Comment (ArtificialOwl) [reply apps/files_external/tests/Controller/AjaxControllerTest.php:144]: i'll clean those call in https://github.com/nextcloud/server/pull/39957

**Pull request merged**